cvmfs/magic_xattr.h

```diff
@@ -0,0 +1,324 @@
+  void Register(const std::string &name, BaseMagicXattr *magic_xattr);
+
+  bool hide_magic_xattrs() { return hide_magic_xattrs_; }
+
+ protected:
+  std::map<std::string, BaseMagicXattr *> xattr_list_;
+  MountPoint *mount_point_;
+  bool hide_magic_xattrs_;
+};
+
+class AuthzMagicXattr : public BaseMagicXattr {
+  virtual bool PrepareValueFenced();
+  virtual std::string GetValue();
+  virtual MagicXattrFlavor GetXattrFlavor();
+};
+
+class CatalogCountersMagicXattr : public BaseMagicXattr {
+  std::string subcatalog_path_;
+  catalog::Counters counters_;
+
+  virtual bool PrepareValueFenced();
+  virtual std::string GetValue();
+};
+
+class ChunkListMagicXattr : public RegularMagicXattr {
+  std::string chunk_list_;
+
+  virtual bool PrepareValueFenced();
+  virtual std::string GetValue();
+};
+
+class ChunksMagicXattr : public RegularMagicXattr {
+  uint64_t n_chunks_;
+
+  virtual bool PrepareValueFenced();
+  virtual std::string GetValue();
+};
+
+class CompressionMagicXattr : public RegularMagicXattr {
+  virtual bool PrepareValueFenced();
+  virtual std::string GetValue();
+};
+
+class ExternalFileMagicXattr : public RegularMagicXattr {
+  virtual bool PrepareValueFenced();
+  virtual std::string GetValue();
+};
+
+class ExternalHostMagicXattr : public BaseMagicXattr {
+  virtual std::string GetValue();
+};
+
+class ExternalTimeoutMagicXattr : public BaseMagicXattr {
+  virtual std::string GetValue();
+};
+
+class FqrnMagicXattr : public BaseMagicXattr {
+  virtual std::string GetValue();
+};
+
+class HashMagicXattr : public WithHashMagicXattr {
+  virtual bool PrepareValueFenced();
+  virtual std::string GetValue();
+};
+
+class HostMagicXattr : public BaseMagicXattr {
+  virtual std::string GetValue();
+};
+
+class HostListMagicXattr : public BaseMagicXattr {
+  virtual std::string GetValue();
+};
+
+class LHashMagicXattr : public WithHashMagicXattr {
+  virtual bool PrepareValueFenced();
+  virtual std::string GetValue();
+};
+
+class NCleanup24MagicXattr : public BaseMagicXattr {
+  virtual std::string GetValue();
+};
+
+class NClgMagicXattr : public BaseMagicXattr {
+  int n_catalogs_;
+
+  virtual bool PrepareValueFenced();
+  virtual std::string GetValue();
+};
+
+class NDirOpenMagicXattr : public BaseMagicXattr {
+  virtual std::string GetValue();
+};
+
+class NDownloadMagicXattr : public BaseMagicXattr {
+  virtual std::string GetValue();
+};
+
+class NIOErrMagicXattr : public BaseMagicXattr {
+  std::string n_io_err_;
+
+  virtual bool PrepareValueFenced();
+  virtual std::string GetValue();
+};
+
+class NOpenMagicXattr : public BaseMagicXattr {
+  virtual std::string GetValue();
+};
+
+class ProxyMagicXattr : public BaseMagicXattr {
+  virtual std::string GetValue();
+};
+
+class PubkeysMagicXattr : public BaseMagicXattr {
+  std::string pubkeys_;
+
+  virtual bool PrepareValueFenced();
+  virtual std::string GetValue();
+};
+
+class RawlinkMagicXattr : public BaseMagicXattr {
```

Comment:
For consistency, this should inherit from a `SymlinkMagicXattr` class, i.e. the class hierarchy should mirror the flavor enum.